packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -465,11 +469,38 @@ export function requestUpdateLane(fiber: Fiber): Lane {
     const currentLanePriority = getCurrentUpdateLanePriority();
     lane = findUpdateLane(currentLanePriority, currentEventWipLanes);
   } else {
-    const schedulerLanePriority = schedulerPriorityToLanePriority(
-      schedulerPriority,
-    );
-
-    lane = findUpdateLane(schedulerLanePriority, currentEventWipLanes);
+    let eventPriority = DefaultEvent;
+    if (enableNativeEventPriorityInference) {
+      eventPriority = getCurrentEventPriority();
+    }
+    switch (eventPriority) {
```

Comment:
Let's reuse the LanePriority type for now so we don't have to translate it again. It doesn't cross a package boundary, and it doesn't leak into userspace, so it's fine that they are coupled.

Eventually `findUpdateLane` is going away and so `getCurrentEventPriority` can return a Lane directly, no extra translation needed.

Reply:
>  It doesn't cross a package boundary, and it doesn't leak into userspace, so it's fine that they are coupled.

Sort of does for third party renderers right? Don’t want to open it up to pass arbitrary lanes. WDYT?

Reply:
Could be `isDiscreteEventContext()` and `isContinuousEventContext()`. Conditionals like that tends to inline well.

Ultimately though we want the compiled output to just be `switch (event.type) { case '...': return 1; case '...' return 1; ... }`

Reply:
You could also export opaque constants from the reconciler like Discrete, Continuous, Default that happen to be internal types so we don't have to convert them

Reply:
Literally exporting them is not so easy because of how the custom reconciler infra is exposed. But I can make them properties on the returned renderer. Similar to where we expose `discreteUpdates` and friends.